File: 2265-partition-array-according-to-given-pivot/2265-partition-array-according-to-given-pivot.java
class Solution {
    public int[] pivotArray(int[] nums, int pivot) {
        int[] ans = new int[nums.length];
        int j = 0;

        for(int num : nums)
            if(num < pivot)
                ans[j++] = num;

        for(int num : nums)
            if(num == pivot)
                ans[j++] = num;

        for(int num : nums)
            if(num > pivot)
                ans[j++] = num;

        return ans;
    }
}

File: 2265-partition-array-according-to-given-pivot/2265-partition-array-according-to-given-pivot.js
/**
 * @param {number[]} nums
 * @param {number} pivot
 * @return {number[]}
 */
var pivotArray = function(nums, pivot) {
    const smaller = [], equal = [], greater = [];
    for(const num of nums){
        if(num < pivot) smaller.push(num);
        else if(num == pivot) equal.push(num);
        else greater.push(num);
    }
    return [...smaller, ...equal, ...greater];
};